support_for_advertising_use_cases.md

```diff
@@ -177,23 +177,23 @@ These use-cases are more focused on the publisher's perspective.
+| [View-through Site Personalization](#view-through-site-personalization) | No support | No support | SPARROW: No support |
+| [Click-through Site Personalization](#click-through-site-personalization) | There should not be any conflict with Chrome's proposed “[Privacy Model for the Web](https://github.com/michaelkleber/privacy-model)” unless this mechanism is used to pass through high-entropy identifiers which could be used to tie user identity across websites. See PING's [Privacy Threat Model](https://w3cping.github.io/privacy-threat-model/#goal-transfer-userid). | Similar to Chrome, this should not fall afoul of Webkit's [Tracking Prevention Policy](https://webkit.org/tracking-prevention-policy/) unless high-entropy identifiers are being used to perform "navigational tracking". See PING's [Privacy Threat Model](https://w3cping.github.io/privacy-threat-model/#goal-transfer-userid). | SPARROW: Possible via URL decoration. If following a SPARROW targeted ad, this URL could technically not contain identifying information except the interest group linked to the winning ad. Personalization can therefore only be done at the publisher origin/interest group level. |
 |[On-site Sponsored Product](#on-site-sponsored-product)| There should not be any conflict with Chrome's proposed “[Privacy Model for the Web](https://github.com/michaelkleber/privacy-model)” | Similar answer as that for Chrome. This should not be in conflict with Webkit's Tracking Prevention Policy given the measurement is entirely within the scope of a single publisher website. | |
 |[Search](#search) | unclear, but relies a lot on contextual | unclear | |
-| [Ad Safety](#ad-safety) | Very limited support as opaque iframe aggregated reporting allow for very little ex post audit. | No Support |  |
-| [Advertisers exclusion](#advertisers-exclusion) | Unclear. To be defined. |  |  |
-| [Revenue management](#revenue-management) | Little support in TURTLEDOVE, as nor reporting nor updating bidding JavaScripts is not real-time. This means delays in both information and action, leading to very complex and inaccurate budget/revenue management. |  |  |
+| [Ad Quality](#ad-quality) | Very limited support as opaque iframe aggregated reporting allow for very little ex post audit. | No Support | SPARROW: The Gatekeeper is reponsible for enforcing the publisher policy. The Gatekeeper could provide a service to ensure Ad Quality before the ad is displayed. <br> The aggregated report and Delayed served ads report can be leveraged for ex post ad quality audit. |
```

Comment:
> SPARROW: The Gatekeeper is reponsible for enforcing the publisher policy.

This statement worries me.  Publishers don't interact with the Gatekeeper; it's a piece of browser-trusted infrastructure.  Surely publishers interact with their own ad network in setting their brand safety policies, and the publisher's ad network is the one responsible for doing the right thing.

The actual execution of the code might happen in the browser (for TURTLEDOVE) or in the Gatekeeper (for SPARROW).  But _responsibility_ isn't being moved around, only which computer is executing the instructions.

> The Gatekeeper could provide a service to ensure Ad Quality before the ad is displayed.

This worries me even more!  It makes it look like you're assigning deep decision-making to the Gatekeeper.  Again, in TURTLEDOVE we're not changing whose code makes a decision, only where that code runs.  If SPARROW isn't sticking to that model, then your explainer needs some much more explicit discussion of this fact.

Reply:
> SPARROW: The Gatekeeper is responsible for enforcing the publisher policy.

Let me rephrase: "the publisher expresses its policy to the gatekeeper so that the Gatekeeper can enforce it."

Indeed, the Gatekeeper only acts on behalf of other actors and merely runs their logic. The gatekeeper doesn't take any decision on its own.
The publisher side (publisher / ad-network) is blind on the interest group/targeting domain. Filtering for ad quality purposed has to be delegated to the entity running the JS (browser/gatekeeper. 

This seems aligned with what is explained in [here](https://github.com/BasileLeparmentier/SPARROW/issues/5#issuecomment-628846717):
> Publisher brand safety could work the same way: the metadata about an interest-group-targeted ad includes its topics (as determined by a server, just like today), publisher controls let them pick what topics are allowed/blocked, and on-device JS compares the two.

Is that correct?

> The Gatekeeper could provide a service to ensure Ad Quality before the ad is displayed.

This sentence refers to something that was mentioned during one the W3C calls. Ads could be "blocked by default" by the gatekeeper before being whitelisted by the publisher-side.
It is only a side note, but it will be removed in the next version, as it may indeed bring some confusion.